### DISCUSSION TITLE
Possible to save the script using the extension?

### DISCUSSION THREAD

**Discussion by derricck** [zero-plusplus/vscode-autohotkey-debug#228] [Ideas]
I have a function that read the script source and append/modify some lines and saves it.

The problem is when I'm debugging the script and I modify something on it and don't save, then when the function read the source and appends new data, the data on the editor will overwrite it when I save.

I would like to request a directive to save the script in case it's being edited.

Scenario:
```c++
;@debug save-script
File := FileOpen(A_ScriptFullPath, "rw")
data := File.Read()

; modify the data and save

File.Length := 0
File.Write(new_data)
File.close()
```
This way is guaranteed to in case the script is being edited on the editor, it gets saved before the function reads and modifies it.

**Comment by zero-plusplus** [comment]
If you want to automatically save the file being edited when debugging starts, set `"allEditorsInActiveGroup"` to `"debug.saveBeforeStart"` in settings.json.

**Reply by derricck** [reply]
Not when debugging start, on my computer vscode already save it whenever I start debugging.

Saving before could occur the scenario I mentioned.

1- The script is being debugged
2- I edit something on it into vscode but don't save (while the script is currently being debugged)
3- The function append new data to script and save
4-  When I save the script being edited on vscode(2), the data appended by the script(3) is overwritten.

This would guarantee the script would be saved before appending something to it:
```ahk
;@debug save-script
File := FileOpen(A_ScriptFullPath, "rw")
data := File.Read()

; modify the data and save

File.Length := 0
File.Write(new_data)
File.close()
```

Isn't possible to call the save from a command into the script similar to my example above?

**Reply by zero-plusplus** [reply]
Sorry. I don't understand what you are trying to do.

Can you provide a specific example?

**Reply by derricck** [reply]
A directive that can save the script case it detect that the script is being edited into vscode
When there's this gray circle icon:

![FE5B606E-6BFD-492A-BF4A-F8260D4E2CC9](https://user-images.githubusercontent.com/105561748/174910282-5ad5b172-ef3c-46a6-8e3c-104ff6ca738f.jpeg)

Something like
`;@debug-save-script`
or if not possible to know it's being edited, just save. 

The example is the scenario I described on my previous comment.

**Reply by zero-plusplus** [reply]
I understand your request, but do not understand why you are doing this.
Why rewrite the file before the script rewrites it?

Also, if an unsaved file is rewritten from the outside, there is a conflict and the file cannot be saved.

Given the information I have now, I don't think implementing the directives is the right solution.
